src/t8_schemes/t8_quads_w_rectangular_subelements/t8_subelements/t8_subelements_quad_cxx.cxx

```diff
@@ -0,0 +1,1523 @@
+  num_subelements = P4EST_FACES + digit_sum_of_binary;
+  return num_subelements;
+}
+
+void
+t8_default_scheme_sub_c::t8_element_get_location_of_subelement (const
+                                                                t8_element_t *
+                                                                elem,
+                                                                int
+                                                                location[])
+{
+  const t8_quad_with_subelements *pquad_w_sub =
+    (const t8_quad_with_subelements *) elem;
+
+  /* this function only works for subelements */
+  T8_ASSERT (pquad_w_sub->dummy_is_subelement == T8_IS_SUBELEMENT);
+
+  T8_ASSERT (t8_element_is_valid (elem));
+
+  /* Consider the following subelement of type 13:
+   *            
+   *              f0                         1
+   *        x - - x - - x              x - - x - - x           
+   *        |           |              | \ 2 | 3 / |           faces:                                                      f3   f2   f1   f0
+   *        |           |              | 1 \ | / 4 |           binary code:                                                 1    1    0    1   (=13)
+   *     f3 x           x f2   -->   1 x - - x - - x 1   -->   rearrange binaries s.t. the faces are enumerated clockwise:  1    1    1    0
+   *        |           |              | 0 /   \ 5 |           number subelements at face:                                  2    2    2    1
+   *        | elem      |              | /   6   \ |           consider sub_id 3:                                                x -> second subelement on the upper face
+   *        + - - - - - x              x - - - - - x
+   *              f1                         0
+   *           
+   * We will use the binary representation to determine the location of the given subelement. 
+   * 
+   * We need to know: 
+   *     i)   the face number of the first vertex (values: {0,1,2,3}).
+   *     ii)  whether this face is split in half (values: {0,1}).
+   *     iii) if the subelement is the first or second subelement at the face (values: {0,1}).
+   * 
+   * These informations are then saved in the location array which will be used by the element_vertex function, 
+   * to automatically determine the vertex coordinates of the given subelement. 
+   * 
+   * The location array for the above example would be {1,1,1} (upper face, split = true, second subelement at the upper face). */
+
+  /* 1) convert the subelement type from a decimal to a binary representation */
+  int                 type = pquad_w_sub->subelement_type;
+  int                 binary_array[P4EST_FACES] = { };
+
+  int                 i;
+
+  for (i = 0; i < P4EST_FACES; i++) {   /* need an array with 4 elements to store all subelement types of the quad scheme from 1 to 15 ({0,0,0,1} to {1,1,1,1}) */
+    if (type >= pow (2, (P4EST_FACES - 1) - i)) {
+      binary_array[i] = 1;
+      type -= pow (2, (P4EST_FACES - 1) - i);
```

Comment:
Use 1<< X instead of pow

Reply:
I use binary operations here, now. We dont need pow anymore.